robot_smach_states/src/robot_smach_states/navigation/find.py

```diff
@@ -43,15 +43,21 @@ def entities_from_description(robot, entity_description, list_of_entity_ids=None
         for id in list_of_entity_ids:
             robot.ed.update_entity(id=id, action='remove')
     else:
-        # Select entities based on the description (only type for now)
-        try:
+        # Select entities based on the description
+        # First case is the old behavior, not nice, but keeping it to be sure nothing breaks TODO: clean this up.
```

Comment:
TODO, so please do.

Reply:
A small improvement has been made to increase the stability of the system, however the code is not perfect yet.